python/ray/dag/tests/test_output_node.py

```diff
@@ -191,6 +191,31 @@ def f(self):
         ray.get(A.remote().f.bind().execute())
 
 
+def test_executes_in_bind_order(shared_ray_instance):
+    """Verify that multiple tasks submitted to the
+    same actor will execute in the order they were
+    bound in
+    """
+
+    @ray.remote
+    class Actor:
+        def foo(self, input_data):
+            return input_data * 2
+
+        def bar(self, input_data):
+            return input_data / 2
+
+    a = Actor.remote()
+
+    with InputNode() as inp:
+        x = a.foo.bind(inp)
+        y = a.bar.bind(inp)
+
+        dag = MultiOutputNode([y, x])
+
+    assert ray.get(dag.execute(4)) == [8, 2]
```

Comment:
`x` would get 8 and `y` would get 2. I would expect executing this dag gives me `[2, 8]` as the result, same as the order specified in `MultiOutputNode`.

Reply:
Hi Andy I had a follow up question about this. I've reworked my solution and I've gotten the functions to execute in bind order without inherently reordering the arguments to a node. However, I've run against the design decision about the ordering in which results should be returned to the user.

`with InputNode() as inp:
    x = a.foo.bind(inp)
    y = a.bar.bind(inp)
    z = a.baz.bind(1, 2, 3, y, inp, x)

    dag = MultiOutputNode([y, x, z])

# Execute the DAG with a sample input and observe the order
print("Starting DAG execution to test task order...")
result = ray.get(dag.execute(4))
print("DAG execution result:", result)`

In the way I wrote my first attempt, when this executes result will get [8, 2, 20] because foo returns 8, bar returns 2 and baz returns 20. Is fundamentally the wrong approach? Should we always return these values in the position that they were passed in? This makes the most intuitive sense to me from a user perspective, but if that's the case then I'm not sure how to write unit tests for the ordering of functions that are executing remotely? Thanks for your help!

Reply:
Yes, we should keep the same order that the values are passed in.

For unit tests, you can check that the order is as expected by modifying state internal to the actor, like keeping a list of tasks that were called.